tracer/src/Datadog.Trace/Debugger/DebuggerManager.cs

```diff
@@ -0,0 +1,343 @@
+
+                if (disabled)
+                {
+                    Log.Information("Exception Replay is disabled by environment variable. To enable it, please set {ExceptionReplayEnabled} environment variable to '1'/'true'.", Datadog.Trace.Configuration.ConfigurationKeys.Debugger.ExceptionReplayEnabled);
+                    return;
+                }
+
+                if (dynamicallyDisabled)
+                {
+                    Log.Information("Exception Replay is disabled by remote enablement. To enable it, re-enable it via Datadog UI");
+                    ExceptionReplay?.Dispose();
+                    ExceptionReplay = null;
+
+                    return;
+                }
+
+                if ((enabled || dynamicallyEnabled) && ExceptionReplay == null)
+                {
+                    var exceptionReplay = ExceptionDebugging.Create(ExceptionReplaySettings);
+
+                    if (exceptionReplay != null)
+                    {
+                        exceptionReplay.Initialize();
+                        ExceptionReplay = exceptionReplay;
+                    }
+                    else
+                    {
+                        Log.Information("Exception Replay is disabled.");
+                    }
+                }
+            }
+            catch (Exception ex)
+            {
+                Log.Error(ex, "Error initializing Exception Replay.");
+            }
+        }
+
+        private async Task SetDynamicInstrumentationState()
+        {
+            try
+            {
+                var enabled = DebuggerSettings.DynamicInstrumentationEnabled.HasValue && DebuggerSettings.DynamicInstrumentationEnabled.Value;
+                var disabled = DebuggerSettings.DynamicInstrumentationEnabled.HasValue && !DebuggerSettings.DynamicInstrumentationEnabled.Value;
+                var dynamicallyEnabled = DebuggerSettings.DynamicSettings.DynamicInstrumentationEnabled.HasValue && DebuggerSettings.DynamicSettings.DynamicInstrumentationEnabled.Value;
+                var dynamicallyDisabled = DebuggerSettings.DynamicSettings.DynamicInstrumentationEnabled.HasValue && !DebuggerSettings.DynamicSettings.DynamicInstrumentationEnabled.Value;
+
+                if (disabled)
+                {
+                    Log.Information("Dynamic Instrumentation is disabled by environment variable. To enable it, please set {DynamicInstrumentationEnabled} environment variable to '1'/'true'.", Datadog.Trace.Configuration.ConfigurationKeys.Debugger.DynamicInstrumentationEnabled);
+                    return;
+                }
+
+                if (dynamicallyDisabled)
+                {
+                    Log.Information("Dynamic Instrumentation is disabled by remote enablement. To enable it, re-enable it via Datadog UI");
+                    TracerManager.Instance.Telemetry.ProductChanged(TelemetryProductType.DynamicInstrumentation, enabled: false, error: null);
```

Comment:
Nit: consider adding telemetry to Exception Replay

Reply:
Will do in a future PR that will handle debugger telemtries